### PR TITLE
feature/EVS-NWPS_regions

### PR DESCRIPTION
**Pull request by SamiraArdani-NOAA**
<b>Note to developers: You must use this PR template!</b>

## Description of Changes

> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

In this PR, the EVS-NWPS codes/scripts were updated to include all WFOs for the stats and plots steps. In the previous PRs for this component, the codes were tested with only one WFO. Now, the scripts have this capability to be run for all WFOs. However, for this test, I included two WFOs to see if the looping over WFOs works. After the successful test of these changes, the other WFOs can be added.
   
## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by?

No.

* Do you have any planned upcoming annual leave/PTO?

No.

* Are there any changes needed for when the jobs are supposed to run?

No.
  
- [x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x] References the feature branch for `HOMEevs` are removed from the code.
- [x] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x] Jobs over 15 minutes in runtime have restart capability.

In this PR, I include two WFOs for testing the loops over the available WFOs. In future developments, I have planned to add all WFOs. Therefore, both "stats" and "plots" steps have a potential to be run in longer times later. So, they might need restart capability. The "stats" step has already this capability. 

- [x] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x] Log is free of any ERRORs or WARNINGs.
No ERRORs. There are WARNINGs for missing data.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

1- To test this PR, you need to run the driver scripts for two steps (stats and plots).
2- Clone my fork in your local and checkout branch feature/EVS-NWPS_regions.
3- ln -s fix directory.
4- To run the stats step, point COMIN to emc.vpppg.
5- Run the driver script for stats step at: EVS/dev/drivers/scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh
6- To test the plots, point COMIN to ${USER}.
7- Run the driver script for plots step at: EVS/dev/drivers/scripts/plots/nwps/exevs_nwps_wave_grid2obs_plots.sh

### CODE REVIEW

File: dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ export HOMEevs=/lfs/h2/emc/vpppg/noscrub/$USER/EVS
 
 export MODELNAME=nwps
 export OBTYPE=NDBC_STANDARD
-export MASK=GOM
+#export MASK=GOM
```
Comment (malloryprow) [inline dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.sh]: If commented line isn't needed, please remove.

File: jobs/JEVS_NWPS_PLOTS [hunk 2]
```diff
@@ -59,6 +59,8 @@ export COMIN=${COMIN:-$(compath.py ${envir}/com/${NET}/${evs_ver})}
 export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})}
 export EVSINnwps=${COMIN}/stats/${COMPONENT}/${MODELNAME}.${VDATE}
 export COMOUTplots=${COMOUT}/${STEP}/${COMPONENT}/${RUN}.${VDATE}
+export COMINnwps=${COMINnwps:-$(compath.py ${envir}/com/${MODELNAME}/${nwps_ver})}
+export COMINhist=${COMINnwps}/${MODELNAME}.${VDATE}/nwges/dcom_hist.txt
```
Comment (malloryprow) [inline jobs/JEVS_NWPS_PLOTS]: When NCO sees `COMINhist`, they'll think it is the com directory for a model name hist. Where is in the code it uses `COMINhist`, can it use `${COMINnwps}/${MODELNAME}.${VDATE}/nwges/dcom_hist.txt` instead?

File: jobs/JEVS_NWPS_STATS [hunk 3]
```diff
@@ -64,15 +64,18 @@ export ARCmodel=${COMIN}/prep/${COMPONENT}
 export COMOUT=${COMOUT:-$(compath.py -o ${NET}/${evs_ver})/${STEP}/${COMPONENT}}
 export COMOUTsmall=${COMOUT}/${RUN}.${VDATE}/${MODELNAME}/${VERIF_CASE}
 export COMOUTfinal=${COMOUT}/${MODELNAME}.${VDATE}
+export COMINhist=${COMINnwps}/${MODELNAME}.${VDATE}/nwges/dcom_hist.txt
```
Comment (malloryprow) [inline jobs/JEVS_NWPS_STATS]: When NCO sees `COMINhist`, they'll think it is the com directory for a model name hist. Where is in the code it uses `COMINhist`, can it use `${COMINnwps}/${MODELNAME}.${VDATE}/nwges/dcom_hist.txt` instead?

File: scripts/plots/nwps/exevs_nwps_wave_grid2obs_plots.sh [hunk 4]
```diff
@@ -56,134 +56,135 @@ echo '-----------------------------'
...
-  EVSINnwps=${COMIN}/stats/${COMPONENT}/${MODELNAME}.${theDate}
-    if [ -s ${EVSINnwps}/evs.stats.${MODELNAME}.${RUN}.${VERIF_CASE}.v${theDate}.stat ]; then
-	    cp ${EVSINnwps}/evs.stats.${MODELNAME}.${RUN}.${VERIF_CASE}.v${theDate}.stat ${DATA}/stats/.
-    else
-	    echo "WARNING: ${EVSINnwps}/evs.stats.${MODELNAME}.${RUN}.${VERIF_CASE}.v${theDate}.stat DOES NOT EXIST"
-    fi
-    theDate=$(date --date="${theDate} + 1 day" '+%Y%m%d')
-done
+WFO='hgx bro'
+WFO=${WFO}
```
Comment (malloryprow) [inline scripts/plots/nwps/exevs_nwps_wave_grid2obs_plots.sh]: Is this line needed `WFO=${WFO}`?

File: scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh [hunk 5]
```diff
@@ -39,10 +39,12 @@ mkdir -p ${DATA}/tmp
 mkdir -p ${DATA}/SFCSHP
 
 vhours='00 06 12 18'
-WFO='hgx'
+WFO='hgx bro'
+WFO=${WFO}
```
Comment (malloryprow) [inline scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh]: Is this line needed `WFO=${WFO}`?

File: scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh [hunk 6]
```diff
@@ -114,70 +122,73 @@ for wfo in ${WFO}; do
...
+		   echo " Found ${nc} ${DATA}/all_stats/${wfo}/*stat files for ${VDATE}"
+		   # Use StatAnalysis to gather the small stat files into one file
+		   run_metplus.py ${PARMevs}/metplus_config/machine.conf ${GRID2OBS_CONF}/StatAnalysis_fcstNWPS_obs$OBSNAME.conf
+		   export err=$?; err_chk
+
+		   if [ $SENDCOM = YES ]; then
+			   if [ -s ${DATA}/stats/${wfo}/evs.stats.${MODELNAME}.${RUN}.${VERIF_CASE}.v${VDATE}.stat ]; then
+				   cp -v ${DATA}/stats/${wfo}/evs.stats.${MODELNAME}.${RUN}.${VERIF_CASE}.v${VDATE}.stat ${COMOUTfinal}/evs.stats.${MODELNAME}.${wfo}.${RUN}.${VERIF_CASE}.v${VDATE}.stat 
+			   else
+				   echo "DOES NOT EXIST ${DATA}/stats/evs.stats.${MODELNAME}.${RUN}.${VERIF_CASE}.v${VDATE}.stat"
```
Comment (malloryprow) [inline scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh]: Should be `${DATA}/stats/${wfo}`, I think

File: scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh [hunk 7]
```diff
@@ -114,70 +122,73 @@ for wfo in ${WFO}; do
...
+			   fi
 		   fi
+	   else
+		   echo "NO SMALL STAT FILES FOUND IN ${DATA}/all_stats"
```
Comment (malloryprow) [inline scripts/stats/nwps/exevs_nwps_wave_grid2obs_stats.sh]: Should be `${DATA}/all_stats/${wfo}`, I think

**Comment by malloryprow** [comment]
I see there is a change to scripts/prep/nwps/exevs_nwps_wave_grid2obs_prep.sh. Should the prep step be included in the testing too?

**Comment by SamiraArdani-NOAA** [comment]
> I see there is a change to scripts/prep/nwps/exevs_nwps_wave_grid2obs_prep.sh. Should the prep step be included in the testing too?

@malloryprow, Ok.

File: jobs/JEVS_NWPS_PLOTS (hunk 2 above)
Comment (SamiraArdani-NOAA) [reply jobs/JEVS_NWPS_PLOTS]: @malloryprow, I had initially added the path to com_hist.txt but later I decided to use that in a later PR when I add a list of all WFOs. So, I removed the line that defines it for now.

**Comment by malloryprow** [comment]
### prep

### 1. jevs_nwps_wave_grid2obs_prep
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/prep/nwps/jevs_nwps_wave_grid2obs_prep.o207164790
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_prep.207164790.dbqs01
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0/prep/nwps/wave

**Comment by SamiraArdani-NOAA** [comment]
> ### prep
> ### 1. jevs_nwps_wave_grid2obs_prep
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/prep/nwps/jevs_nwps_wave_grid2obs_prep.o207164790 **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_prep.207164790.dbqs01 **COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0/prep/nwps/wave

The prep looks good. The .log file is free from ERRORs. I see that one of the WFOs (mfl) now has been added to the outputs as a result of this name correction in prep.

**Comment by malloryprow** [comment]
### stats

### 1. jevs_nwps_wave_grid2obs_stats
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/stats/nwps/jevs_nwps_wave_grid2obs_stats.o207166687
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_stats.207166687.dbqs01
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0/stats/nwps

**Comment by SamiraArdani-NOAA** [comment]
> ### stats
> ### 1. jevs_nwps_wave_grid2obs_stats
> **Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/stats/nwps/jevs_nwps_wave_grid2obs_stats.o207166687 **DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_stats.207166687.dbqs01 **COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0/stats/nwps

The stats looks as expected. No error was found in the .log file. Also, I compared the results for wfo=hgx with those of emc.vpppg and they are identical. The outputs for the other wfo in this test looks good too.

**Comment by malloryprow** [comment]
Great! For the plots should I point to the emc.vpppg parallel? Or do you have an archive? If I keep COMIN set to $USER, I won't have any stats.

**Comment by SamiraArdani-NOAA** [comment]
> Great! For the plots should I point to the emc.vpppg parallel? Or do you have an archive? If I keep COMIN set to $USER, I won't have any stats.

I have an archive for the last two~three days that I was testing stats. These outputs are located at: /lfs/h2/emc/vpppg/noscrub/samira.ardani/evs/v2.0/stats/nwps/

**Comment by malloryprow** [comment]
### plots

### 1. jevs_nwps_wave_grid2obs_plots
**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.o207169718
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_plots.207169718.dbqs01
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0

**Comment by SamiraArdani-NOAA** [comment]
> /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.o207169718
I had to copy your recent stats runs for 20241121 to my archive before you run the plots. Would you please run the plots one more time now? Sorry about it.

**Comment by malloryprow** [comment]
Rerun!

### jevs_nwps_wave_grid2obs_plots
Log File: /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.o207170585
DATA: /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_plots.207170585.dbqs01
COMOUT: /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0

**Comment by SamiraArdani-NOAA** [comment]
> Rerun!
> 
> ### jevs_nwps_wave_grid2obs_plots
> Log File: /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.o207170585 DATA: /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_plots.207170585.dbqs01 COMOUT: /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0

@malloryprow, Thanks for rerunning. I think your COMIN is still pointing to emc.vpppg. Would you please change it to my directory.

**Comment by malloryprow** [comment]
> /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0

Third time is the charm...I didn't have COMIN right the second time

**Log File:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/EVS/dev/drivers/scripts/plots/nwps/jevs_nwps_wave_grid2obs_plots.o207171169
**DATA:** /lfs/h2/emc/stmp/mallory.row/evs_test/prod/tmp/jevs_nwps_wave_grid2obs_plots.207171169.dbqs01
**COMOUT:** /lfs/h2/emc/vpppg/noscrub/mallory.row/verification/EVS_PRs/pr609/evs/v2.0

**Comment by SamiraArdani-NOAA** [comment]
@malloryprow, Thanks for rerunning. Looks like the plots step is working. I did not find errors and the outputs look as expected.

**Review by malloryprow — APPROVED** [review]
Changes are good and testing successful.

**Review by AndrewBenjamin-NOAA — APPROVED** [review]
I approve the changes to this PR. This is another great step toward getting all WFO's into the NWPS component, Thank you!

**Pull request merged**